interview-questions.md

```diff
@@ -7,53 +7,170 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: 
+
+Rails is an open source web application framework which runs the Ruby programming language. Rails is written in Ruby. 
+It allows for writing less code and the Rails philosphpy is to not repeat yourself and a set of conventions on the best way to do something. 
+Rails has a prebuilt structure for development. 
+
+Rails is like pre built "Legos" of code that you can mix, match, and modify to create a custom web application. You don't have to build 
+everything from scratch. 
+
+Ruby on Rails uses the Model-View Constroller (MVC) architectural pattern used by many other web frameworks. 
+MVC pattern separates web apps code into three parts. 
+
+The Model-holds the data structure
+The View- the visual part of web page templates
+THe Controller- conects the View and the logic of the app
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database contains tables with rows and columns. 
 
-  Researched answer:
+  Researched answer: A relational databse is a collection of information that is organized into data points. They have been around since the 
+  1970's. Structured Query Language SQL is the most common for readding, creating, and updating, and deleting data. Relational databases are 
+  very reliable. They are organzied as a set of tables with columns and rows. Each column hodls a certain type of data and field 
+ that has an actual value of an attribute. The rows in the table represent a collection of related values 
+ Each row in a table could be marked with a unquie idenifier, a primayr key. 
 
+ Examples: Microsoft SQL Server, Oracle Database, MySQL, PostgreSQL and IBM Db2
 
+Non-relational databases are those that do not use SQL as their primary data access language and are referred to as NoSQL databases.
+Because data in a NoSQL database does not have to comply to a pre-defined schema, unlike data in relational databases, these databases are ideal for storing unstructured or semi-structured data.
+One benefit of NoSQL databases is that they allow developers to make changes to the database on the fly without disrupting apps that use it.
 
-5. What are primary keys? Why are they important?
+ Examples: Apache Cassandra, MongoDB, CouchDB, and CouchBase
 
-  Your answer:
+Any database that is built to function in the cloud is referred to as a cloud database.
+Cloud databases, like other cloud-based applications, provide flexibility, scalability, and high availability.
+Because many cloud databases are provided as a service, they are frequently low-maintenance.
 
-  Researched answer:
 
+Examples: Microsoft Azure SQL Database, Amazon Relational Database Service, Oracle Autonomous Database.
+
+
+5. What are primary keys? Why are they important?
+
+  Your answer: Primary Keys are the listed in the column and important because they contain the values. 
```

Comment:
This is the kind of answer I like to see in your answers!  Defines to the best of your knowledge and tries to tie it to something you have worked with.